### ISSUE TITLE
Character story progress (quests) doesn't reflect the in-game completion state properly

### ISSUE THREAD

**Issue by Archomeda** [gw2-api/issues#16]
### Current Behavior

Currently, the API tells me (/v2/characters/:id/quests) that one of my characters doesn't have the following quests completed, despite that I have definitely completed them on this character:
- HoT 409 - Torn from the Sky
- HoT 410 - Prisoners of the Dragon
- HoT 404 - Hearts and Minds
- IBS 581 - Roaring Flames
- IBS 582 - Defending Tyria
- IBS 583 - Defending Tyria
- IBS 587 - Calm in the Storm
- IBS 590 - Flame vs. Frost
- IBS 593 - Frozen Tide
- IBS 595 - Primordus Rising
- IBS 596 - Tidings and Misgivings
- IBS 598 - Champion's End
- IBS 599 - Wildfire
- IBS 600 - Calm in the Storm
- IBS 602 - Calm in the Storm
- IBS 605 - Jormag's World
- IBS 606 - Calm in the Storm
- IBS 608 - Jormag's Madness
- IBS 611 - Dragonstorm

### Expected Behavior

The ids of completed quests should be included in /v2/character/:id/quests.

### Steps To Reproduce

1. Go to [/v2/characters/:id/quests for one of my characters](https://api.guildwars2.com/v2/characters/Reiga%20Fiercecrusher/quests?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI5WjZ1SElkNnpPbmFJSzJKNE5Ma3BUcTUxRkNSc3NOVHB5cGtURHhiVFI4IiwiaWF0IjoxNjM2NjYzMzIyLCJleHAiOjE2NjgxOTkzMjIsInBlcm1pc3Npb25zIjpbImFjY291bnQiLCJjaGFyYWN0ZXJzIiwicHJvZ3Jlc3Npb24iXSwidXJscyI6WyIvdjIvY2hhcmFjdGVycy9SZWlnYSUyMEZpZXJjZWNydXNoZXIvcXVlc3RzIl19.RK-yfTYhstnGNpRwuPbdRe8EQSfshpYZ2ylYZW7upss&v=latest)
2. Observe that the aforementioned ids are not included (as of 2021-11-11, it may magically fix itself somehow later)

### Happening since

Unknown

### Anything else?

Quest id 569 is currently also not included in the list, but I've left that one out from this issue, because I have it currently in the replay state, which might explain why it's also not listed as completed under /v2/characters/:id/quests, however the others are not in any replay state.

**Comment by Sounomi** [comment]
It's been a known issue since this end-point was added that if you replay an episode but don't finish it completely, all subsequent quests will show up as not being complete.  Perhaps this isn't a bug but rather an oversight and I think a good fix would be to simply add somewhere in the end-point which story episodes/chapters have been completed, upon which we could just ignore the quest status for that episode/chapter.

In my personal experience, everything quest shows up as complete unless I hit replay for that episode/chapter.  The mere act of hitting replay with reset the quest status for it, even if you don't actually replay it or set foot in any of its instances.

**Comment by Archomeda** [comment]
> It's been a known issue since this end-point was added that if you replay an episode but don't finish it completely, all subsequent quests will show up as not being complete.

Fair point. I would classify this as a separate issue then of which the cause is known. I rechecked the list, and all IBS story quests are now listed again, so that's probably the cause.

However, the HoT ones are still missing for me, even though they are finished and not in a replay state:
- HoT 409 - Torn from the Sky
- HoT 410 - Prisoners of the Dragon
- HoT 404 - Hearts and Minds

**Comment by martingjohn** [comment]
I've got a similar issue with End of Dragons, where I have the last 4 quests of Act 1 that aren't marked as complete 

629 - Old Friends
622 - Outreach
614 - Your Kind of People
620 - Can't Trust a Pirate

and neither is the 1st quest of Act 2

626 - The Scenic Route

but all the ones after that are.

Similarly, in Secrets of the Obscure, I'm missing 702 - 11. Duress, but all the ones after that are there.
